manager/state/raft/storage/nacl.go

```diff
@@ -0,0 +1,74 @@
+package storage
+
+import (
+	"crypto/rand"
+	"fmt"
+	"io"
+
+	"golang.org/x/crypto/nacl/secretbox"
```

Comment:
Full transparency/disclosure: I am a cryptographer, I have [a libsodium binding](https://github.com/lvh/caesium), and I actively lobbied @cyli in favor of NaCl.